roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -105,6 +105,31 @@ def get_var(self, *keys, **kwargs):
             raise OpenShiftCheckException("'{}' is undefined".format(".".join(map(str, keys))))
         return value
 
+    def find_ansible_mount(self, path):
+        """Return the mount point for path from ansible_mounts."""
+
+        # reorganize list of mounts into dict by path
+        mount_for_path = {
+            mount['mount']: mount
+            for mount
+            in self.get_var('ansible_mounts')
+        }
+
+        mount_point = path
+        while mount_point not in mount_for_path:
```

Comment:
while + condition + if + break == while + new_condition

Let's see...

```python
while mount_point not in mount_for_path and mount_point not in ["/", ""]:
    mount_point = os.path.dirname(mount_point)
```

Simplifying:

```python
while mount_point not in list(mount_for_path.keys()) + ["/", ""]:
    mount_point = os.path.dirname(mount_point)
```

Since the condition is re-evaluated in every iteration, and to make it shorter:

```python
# NOTE: we add '/' and '' because those are the base paths that
# os.path.dirname might return, preventing an infinite loop.
mount_points = set(mount_for_path.keys()) | {'/', ''}
while mount_point not in mount_points:
    mount_point = os.path.dirname(mount_point)
```

Hmm, the single condition still sort of asks for an explanation why we need the extra paths :-)